src/OAuthClient/HoldRequestClient.php

```diff
@@ -118,25 +154,55 @@ public static function patchHoldRequestById($holdRequestId = '', bool $processed
 
         $url = Config::get('API_HOLD_REQUEST_URL') . '/' . $holdRequestId;
 
-        APILogger::addDebug('Patching hold request by id', (array) $url);
-
         $body = ["processed" => $processed, "success" => $success];
 
-        $response = self::patch($url, ["body" => json_encode($body)]);
+        APILogger::addDebug('Patching hold request by id', array($url, $body));
 
-        $response = json_decode((string)$response->getBody(), true);
+        try {
+            $response = self::patch($url, ["body" => json_encode($body)]);
 
-        APILogger::addDebug('Patched hold request by id', $response['data']);
+            $statusCode = $response->getStatusCode();
 
-        if ($response['statusCode'] !== 200) {
-            APILogger::addError(
-                'Failed',
-                array('Failed to retrieve Hold Request ', $holdRequestId, $response['type'], $response['message'])
-            );
-            return null;
-        }
+            $response = json_decode((string)$response->getBody(), true);
 
+            APILogger::addDebug('Patched hold request by id', $response['data']);
 
-        return new HoldRequest($response['data']);
+            // Check statusCode range
+            if ($statusCode === 200) {
+                return new HoldRequest($response['data']);
+            } else {
+                APILogger::addError(
+                    'Failed',
+                    array('Failed to retrieve Hold Request ', $holdRequestId, $response['type'], $response['message'])
+                );
+                return null;
+            }
+        } catch (ServerException $exception) {
+            throw new RetryableException(
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.